tests/platform_helper/providers/test_ecr.py

```diff
@@ -97,150 +113,158 @@ def two_pages_of_describe_repository_data():
-    ecr = ECRProvider(Mock(), Mock())
+def test_get_commit_tag_for_reference_errors_when_no_images_match(reference):
+    mocks = ECRProviderMocks()
+    mocks.client_mock.list_images.return_value = image_id_pages["page_3"]
+    ecr_provider = ECRProvider(**mocks.params())
 
-    actual_tag = ecr.find_commit_tag(image_details, "commit-abc123")
+    with pytest.raises(ImageNotFoundException) as ex:
+        ecr_provider.get_commit_tag_for_reference("test_app", "test_codebase", reference)
 
-    ecr.click_io.info.assert_not_called()
-    ecr.click_io.warn.assert_not_called()
-    assert actual_tag == "commit-abc123"
+    actual_error = str(ex.value)
+    expected_error = IMAGE_NOT_FOUND_TEMPLATE.format(image_ref=reference)
 
+    assert actual_error == expected_error
 
-def test_find_commit_tag_returns_the_original_ref_if_no_commit_tag():
-    ecr = ECRProvider(Mock(), Mock())
 
-    image_details = [{"imageTags": ["tag-1.2.3", "branch-main"]}]
+def test_get_commit_tag_for_reference_errors_when_multiple_images_match():
+    mocks = ECRProviderMocks()
+    mocks.client_mock.list_images.return_value = image_id_pages["page_3"]
+    ecr_provider = ECRProvider(**mocks.params())
 
-    actual_tag = ecr.find_commit_tag(image_details, "tag-1.2.3")
+    with pytest.raises(MultipleImagesFoundException) as ex:
+        ecr_provider.get_commit_tag_for_reference("test_app", "test_codebase", "commit-deadbea7")
 
-    ecr.click_io.warn.assert_called_once_with(
-        'WARNING: The AWS ECR image "tag-1.2.3" has no associated commit tag so deploying "tag-1.2.3". Note this could result in images with unintended or incompatible changes being deployed if new ECS Tasks for your service.'
+    actual_error = str(ex.value)
+    expected_error = MULTIPLE_IMAGES_FOUND_TEMPLATE.format(
+        image_ref="commit-deadbea7", matching_images="commit-dea, commit-deadb, commit-deadbea"
     )
-    assert actual_tag == "tag-1.2.3"
+
+    assert actual_error == expected_error
 
 
 @pytest.mark.parametrize(
-    "image_details",
+    "boto_exception, expected_exception, expected_message",
     [
-        None,
-        [],
-        [{}],
-        [{"imageDetails": None}],
-        [{"imageDetails": []}],
+        (
+            "RepositoryNotFoundException",
+            RepositoryNotFoundException,
+            REPOSITORY_NOT_FOUND_TEMPLATE.format(repository="test_app/test_codebase"),
+        ),
+        (
+            "SomeOtherException",
+            AWSException,
+            "Unexpected error for repo 'test_app/test_codebase' and image reference 'commit-abc123': "
+            "An error occurred (SomeOtherException) when calling the ListImages operation: Unknown",
+        ),
     ],
 )
-def test_find_commit_tag_handles_malformed_image_details(image_details):
-    ecr = ECRProvider(Mock(), Mock())
+def test_get_commit_tag_for_reference_recasts_exceptions_as_platform_exceptions(
```

Comment:
Just wondering given the test name and intention, should we catch the exception as a `PlatformException` instead of using the 'expected_exception' parameter?

Reply:
How about renaming the tests as:
```suggestion
def test_get_commit_tag_for_reference_recasts_exceptions_as_more_specific_exceptions(
```